src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -258,6 +260,9 @@ get_retention_policy(const storage::ntp_config::default_overrides& prop) {
         //
         // This differs from ordinary storage GC, in that we are applying
         // space _or_ time bounds: not both together.
+        //
+        // This will also drop the compact settings and replace it with
+        // delete.
```

Comment:
Should we tweak this to preserve the original cleanup policy?

Reply:
we should, but in a different PR
in this PR the behaviour stays the same as much as possible to keep it backportable